engines/wasmer/wasmer.go

```diff
@@ -95,28 +97,43 @@ type (
 var _ = (wapc.Module)((*Module)(nil))
 var _ = (wapc.Instance)((*Instance)(nil))
 
-var engineInstance = engine{}
+// EngineOption is a wasmer-specific configuration, applied via Engine
+type EngineOption func(e *engine)
 
-func Engine() wapc.Engine {
-	return &engineInstance
+// WithEngine allows you to override the default engine. Defaults to
+// `wasmer.NewEngine`.
+func WithEngine(engine *wasmer.Engine) *wasmer.Engine {
+	return engine
 }
 
+func Engine(engs ...*wasmer.Engine) wapc.Engine {
```

Comment:
I think you meant to use the func type

```suggestion
func Engine(options ...EngineOption) wapc.Engine {
```

Reply:
I thought I had it as such and there was feedback from either @pkedy or @syrusakbary that had me switch to the specific engine*.  I can put it back; I just want to make sure we don't go back and forth. Thanks

Reply:
yep. if the preference is to remove the option type I think it still exists in the file, so should also be removed from there.

Reply:
I see.  Doing so now.

Reply:
Sorry ran out of time today.  Will commit hopefully Sat night or Sunday.

Reply:
@codefromthecrypt Sorry about the delay; I was rushing on Friday night.  I think the change was actually very small.  Let me know if this is not what you were asking for.